libdnf/module/ModulePackageContainer.cpp

```diff
@@ -691,16 +710,44 @@ ModulePackageContainer::Impl::moduleSolve(const std::vector<ModulePackage *> & m
     dnf_sack_make_provides_ready(moduleSack);
     Goal goal(moduleSack);
     Goal goalWeak(moduleSack);
-    for (const auto &module : modules) {
-        std::ostringstream ss;
-        auto name = module->getName();
-        ss << "module(" << name << ":" << module->getStream() << ")";
-        Selector selector(moduleSack);
-        bool optional = persistor->getState(name) == ModuleState::DEFAULT;
-        selector.set(HY_PKG_PROVIDES, HY_EQ, ss.str().c_str());
+
+    std::set<std::pair<std::string, std::string>> module_packages_already_requested;
```

Comment:
Be honest I am curious whether this is a good option to use a pair instead of a single string in format `name`:`stream`. Anyway we use that string to create provide -> `ss << "module(" << name << ":" << stream << ")";`.

Reply:
OK sure, I've changed it to a single string.